frappe/core/doctype/user/user.py

```diff
@@ -590,6 +590,13 @@ def on_trash(self):
 					note.remove(row)
 			note.save(ignore_permissions=True)
 
+		# Unlink user from all of its invitation docs
+		invites = frappe.db.get_all("User Invitation", filters={"email": self.name}, pluck="name")
+		for invite in invites:
+			invite_doc = frappe.get_doc("User Invitation", invite)
+			invite_doc.user = None
+			invite_doc.save(ignore_permissions=True)
```

Comment:
Just ignore the links to `User Invitation` instead of changing data.

Reply:
https://github.com/frappe/frappe/blob/0cd2f9ef055f8207df9ece79825269f509c0d7bc/frappe/hooks.py#L395-L396